boring/src/ssl/mod.rs

```diff
@@ -1707,6 +1715,39 @@ impl SslContextBuilder {
         }
     }
 
+    fn set_default_curves(&mut self) -> Result<(), ErrorStack> {
+        cfg_if::cfg_if! {
+            if #[cfg(feature = "pq-preferred")] {
+                self.set_curves(&[
+                    #[cfg(not(feature = "fips-required"))]
+                    SslCurve::X25519_KYBER512_DRAFT00,
+                    SslCurve::P256_KYBER768_DRAFT00,
+                    #[cfg(not(feature = "fips-required"))]
+                    SslCurve::X25519,
+                    SslCurve::SECP256R1,
+                    SslCurve::SECP384R1,
+                ])?;
+            } else if #[cfg(feature = "pq-supported")] {
```

Comment:
Does that mean `pq-preferred` and `pq-supported` are mutually exclusive?

Reply:
`pq-preferred` implies `pq-supported`: https://github.com/cjpatton/boring/blob/pq-by-default/boring/Cargo.toml#L42

So if `pq-preferred` is set but `pq-supported` is, the `else if` will be taken here. Otherwise, if `pq-preferred` is set, then the `if` above will be taken. (This is what we want.)